src/components/SongQueueItem/SongQueueItem.tsx

```diff
@@ -2,60 +2,73 @@ import React from 'react';
 import './SongQueueItem.css';
 import { ListItem, Artist, Song } from '../../assests/globalStyles';
 import styled from 'styled-components';
-import { Icon } from 'antd';
+import { Icon, Modal } from 'antd';
 import { SongQueue, CurrentSong } from '../../types';
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
+import { updateSongDiamonds, getUserProfile } from '../../redux/actions'
+const { confirm } = Modal;
 
 const DiamondIcon = styled(Icon)`
-  font-size: 30px;
-  padding: 0 10px;
-  svg {
-    fill: var(--secondary-color);
-  }
+font-size: 30px;
+padding: 0 10px;
+svg {
+  fill: var(--secondary-color);
+}
 
-  &:hover {
-    opacity: .8;
+&:hover {
+  opacity: .8;
 }`;
 
 const StarIcon = styled(Icon)`
-  font-size: 35px;
-  padding: 0 5px;
-  svg {
-    fill: var(--secondary-color);
-  }
+font-size: 35px;
+padding: 0 5px;
+svg {
+  fill: var(--secondary-color);
+}
 }`;
 
 const StyledArtist = styled(Artist)`
-  color: var(--fourth-color);
-  font-size: 14px;
-  font-weight: normal;
+color: var(--fourth-color);
+font-size: 14px;
+font-weight: normal;
 `;
 
 const StyledSong = styled(Song)`
-  font-size: 14px;
-  font-weight: normal;
+font-size: 14px;
+font-weight: normal;
 `;
 
 const ColContainer = styled.div`
-  display: flex;
-  flex-direction: column;
-  justify-content: flex-start;
+display: flex;
+flex-direction: column;
+justify-content: flex-start;
 `;
 
 const RowContainer = styled.div`
-  display: flex;
-  justify-content: flex-start;
-  align-items: center;
+display: flex;
+justify-content: flex-start;
+align-items: center;
 `;
 
-
-
 const SongQueueItem: React.FC<{ songQueueItem: SongQueue }> = ({ songQueueItem }) => {
+  
+  const dispatch = useDispatch();
 
   const song: CurrentSong = JSON.parse(songQueueItem.song);
   const songUserEmail = songQueueItem.user_id;
   const userEmail = useSelector((state: any) => state.user.userProfile.email)
-
+  const accessToken = useSelector((state: any) => state.user.accessToken)
+  
+  const showConfirm = () => {
+    confirm({
+      title: `Do you want to spend 5 diamonds to promote this song?`,
+      onOk() {
+        songQueueItem.id && dispatch(updateSongDiamonds(songQueueItem.id, accessToken))
+        dispatch(getUserProfile());
+      }
+    });
+  }
+  
```

Comment:
Is it possible to display the targeted song in the modal so the user can check if he selected the right one?
`Do you want to spend 5 diamonds to promote ${song.title} by ${song.artist}?`

Reply:
done